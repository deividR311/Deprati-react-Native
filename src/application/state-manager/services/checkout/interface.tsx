import { DeliveryOptionsResponse } from '../../../../infrastucture/apis/delivery-address';
import { Cart } from '../../../../infrastucture/apis/shopping-cart';
import { CheckoutSteps } from '../../../../presentation/navigation/checkout/interfaces';

export interface CheckoutStateHook {
  onContinueButtonTrigger?: CheckoutSteps;
  onPressContinueButton: (screenId: CheckoutSteps) => void;
}

export interface CheckoutState {
  screenId?: CheckoutSteps;
  previousScreen?: CheckoutSteps;
  stackSummary?: any[];
  showLoadingScreen?: boolean;
  onClickedContinueButton: boolean;
  cart: Partial<Cart>;
  totalPriceBeforeSelectDeliveryMode?: string;
  needRefreshCart?: boolean;
  deliveryOptions?: DeliveryOptionsResponse;
  isGiftCardCart: boolean;
  creditInfo?: {
    saveForFuturePay: boolean;
    maskedPhoneNumber: string;
    maskedNumberAccount: string;
    maskedEmail: string;
    directCreditCodeCustomer?: string;
    directCreditCodeAdditional?: string;
  };
  firstTimeSummary?: boolean;
}

export const INITIAL_STATE: CheckoutState = {
  screenId: undefined,
  previousScreen: undefined,
  stackSummary: undefined,
  onClickedContinueButton: false,
  showLoadingScreen: false,
  needRefreshCart: false,
  cart: {},
  deliveryOptions: undefined,
  totalPriceBeforeSelectDeliveryMode: undefined,
  isGiftCardCart: false,
  creditInfo: undefined,
  firstTimeSummary: undefined
};
