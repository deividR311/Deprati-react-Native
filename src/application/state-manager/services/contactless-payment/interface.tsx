import { ContactLessPaymentSteps } from '../../../../presentation/navigation/contactless-payment';

export enum BottomSheetType {
  IDENTITY_VALIDATION = 'IDENTITY_VALIDATION',
  SUCCESS_BUY = 'SUCCESS_BUY'
}

export interface ShowBottomSheetAction {
  bottomSheet: BottomSheetType;
  show: boolean;
  text?: string;
  ticketId?: string;
}

export interface ContactlessPaymentStateHook {
  TicketNumberContactLessPayment?: string;
  showIdentityValidation(show: boolean): void;
  identityValidationIsOpen: boolean;
  showSuccessBuy(show: boolean, text?: string, ticketId?: string): void;
  successBuyIsOpen: boolean;
  successBuyText?: string;
  successBuyTicketId?: string;
  doEnableConfirmPurchaseButton(isEnable: boolean, ticketId?: string): void;
  enableConfirmPurchaseButton: boolean;
}

export interface ContactlessPaymentState {
  tabbar: {
    disableNextButton: boolean;
    disablePreviousButton: boolean;
    currentTab: ContactLessPaymentSteps | '';
  };
  bottomSheetState: {
    identityValidationIsOpen: boolean;
    successBuyIsOpen: boolean;
    successBuyText: string;
    successBuyTicketId: string;
  };
  enableConfirmPurchaseButton: boolean;
}

export const INITIAL_STATE: ContactlessPaymentState = {
  tabbar: {
    currentTab: '',
    disableNextButton: false,
    disablePreviousButton: false
  },
  bottomSheetState: {
    identityValidationIsOpen: false,
    successBuyIsOpen: false,
    successBuyText: '',
    successBuyTicketId: ''
  },
  enableConfirmPurchaseButton: false
};
