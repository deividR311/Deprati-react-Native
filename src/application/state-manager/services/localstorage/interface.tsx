import { NotificationListRes } from '../../../../infrastucture/apis/notifications/notifications.interfaces';
import { MessagesAppInterface } from '../../../../presentation/screens/Splash/hooks/messages.interface';
import { INITIAL_STATE as ContactlessInitialState } from '../contactless-payment';
import {
  UserResponse,
  USER_DEFAULT_DATA
} from '../../../../infrastucture/apis/user/user.interface';

export type LocalStorageParams = Partial<LocalStorageType>;

export interface LocalStorageHook {
  localStorageData: LocalStorageType;
  save(params: LocalStorageParams): void;
  remove(keys: LocalStorageKey[]): void;
  readAndStorageInState(): void;
  fullClean(): void;
}

export enum LocalStorageKey {
  // BEHAVIOR
  WasStoreRead = 'WAS_STORE_READ',
  isAppOpenByFirstTime = 'IS_APP_OPEN_BY_FIRST_TIME',
  IpAddress = 'IP_ADDRESS',
  UID = 'DEVICE_ID',
  IsEnableConfirmPurchaseButtonInQRScreen = 'CONTACTLESS_PAYMENT@IS_ENABLE_CONFIRM_PURCHASE_BUTTON',
  TicketNumberContactLessPayment = 'CONTACTLESS_PAYMENT@TICKET_NUMBER',
  AnonymousCartGuid = 'ANONYMOUS_CART_GUID',

  // USER DATA
  User = 'USER_DATA',
  UserEmail = 'USER_EMAIL',
  UserLastEntry = 'USER_LAST_ENTRY',

  // ACCOUNT DATA
  IsAccountAuthenticated = 'IS_ACCOUNT_AUTHENTICATED',
  AccountNumber = 'ACCOUNT_NUMBER',
  AccountDisplayNumber = 'ACCOUNT_DISPLAY_NUMBER',
  AccountAdditionalNumber = 'ACCOUNT_ADDITIONAL_NUMBER',
  AccountPhoneNumber = 'ACCOUNT_DISPLAY_PHONE_NUMBER',
  LastCustomerId = 'LAST_CUSTOMER_ID',

  //Notifications
  Notifications = 'NOTIFICATIONS',

  //Notifications Push FCM
  TokenFCM = 'TOKENFCM',

  // SECURITY
  Token = 'TOKEN',
  RefreshToken = 'REFRESH_TOKEN',
  ExpiresIn = 'EXPERES_IN',
  ExpiresAt = 'EXPERES_AT',
  GetTokenDate = 'GET_TOKEN_DATE',
  IsLogin = 'IS_LOGIN',
  IsSucessFirstLogin = 'IS_SUCCESS_FIRST_LOGIN',

  BiometricPublicKey = 'publicKey-generate',
  MessagesApp = 'MESSAGES_APP',
  LocalStorageKey = 'LocalStorageKey',
  Whatsapp = 'WHATSAPP'
}

export interface LocalStorageType {
  [LocalStorageKey.WasStoreRead]: boolean;
  [LocalStorageKey.isAppOpenByFirstTime]: boolean;
  [LocalStorageKey.UserLastEntry]: string;
  [LocalStorageKey.UserEmail]: string;
  [LocalStorageKey.AccountNumber]: string;
  [LocalStorageKey.AccountDisplayNumber]: string;
  [LocalStorageKey.AccountAdditionalNumber]: string;
  [LocalStorageKey.AccountPhoneNumber]: string;
  [LocalStorageKey.LastCustomerId]: string;
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]: boolean;
  [LocalStorageKey.TicketNumberContactLessPayment]?: string;
  [LocalStorageKey.AnonymousCartGuid]?: string;
  [LocalStorageKey.User]: UserResponse;
  [LocalStorageKey.Token]: string;
  [LocalStorageKey.RefreshToken]: string;
  [LocalStorageKey.ExpiresIn]: number;
  [LocalStorageKey.ExpiresAt]: string;
  [LocalStorageKey.GetTokenDate]: string;
  [LocalStorageKey.IsLogin]: boolean;
  [LocalStorageKey.IsAccountAuthenticated]: boolean;
  [LocalStorageKey.IpAddress]: string;
  [LocalStorageKey.BiometricPublicKey]: string;
  [LocalStorageKey.UID]: string;
  [LocalStorageKey.IsSucessFirstLogin]: boolean;
  [LocalStorageKey.Notifications]: {
    unread: number;
    notifications: NotificationListRes[];
  };
  [LocalStorageKey.MessagesApp]: MessagesAppInterface;
  [LocalStorageKey.TokenFCM]: string;
  [LocalStorageKey.Whatsapp]: {
    message: string;
    phone: string;
  };
}

export const InitialLocalStorage: LocalStorageType = {
  [LocalStorageKey.WasStoreRead]: false,
  [LocalStorageKey.isAppOpenByFirstTime]: true,
  [LocalStorageKey.AccountNumber]: '',
  [LocalStorageKey.AccountDisplayNumber]: '',
  [LocalStorageKey.AccountAdditionalNumber]: '',
  [LocalStorageKey.AccountPhoneNumber]: '',
  [LocalStorageKey.LastCustomerId]: '',
  [LocalStorageKey.User]: USER_DEFAULT_DATA,
  [LocalStorageKey.UserEmail]: '',
  [LocalStorageKey.UserLastEntry]: '',
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]:
    ContactlessInitialState.enableConfirmPurchaseButton,
  [LocalStorageKey.TicketNumberContactLessPayment]: '',
  [LocalStorageKey.AnonymousCartGuid]: '',
  [LocalStorageKey.Token]: '',
  [LocalStorageKey.RefreshToken]: '',
  [LocalStorageKey.ExpiresIn]: 0,
  [LocalStorageKey.ExpiresAt]: '',
  [LocalStorageKey.GetTokenDate]: '',
  [LocalStorageKey.IsLogin]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IpAddress]: '0.0.0.0',
  [LocalStorageKey.BiometricPublicKey]: '',
  [LocalStorageKey.UID]: '',
  [LocalStorageKey.IsSucessFirstLogin]: false,
  [LocalStorageKey.Notifications]: {
    unread: 0,
    notifications: []
  },
  [LocalStorageKey.MessagesApp]: {},
  [LocalStorageKey.TokenFCM]: '',
  [LocalStorageKey.Whatsapp]: {
    phone: '',
    message: ''
  }
};
