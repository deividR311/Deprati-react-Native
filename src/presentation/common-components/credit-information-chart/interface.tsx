export interface CreditInformationChartProps {
  ownerFullname?: string;
  cardNumber?: string;
  affiliateDate?: string;
  totalAmount?: number;
  amountSpent?: number;
  availableAmount?: number;
  numeroTarjetaAdicionalDisplay?: string;
  showAffiliateDate?: boolean;
  showDisplayCard?: boolean;
  displayCartType?: 'small' | 'large';
  invertValues?: boolean;
  constentChart: any;
  wrapperChartInCard?: boolean;
}

export interface DonutChartProps {
  indicatorValue?: number;
  maxIndicatorValue?: number;
  color?: string;
  dialColor?: string;
  radius?: number;
  strokeWidth?: number;
  duration?: number;
  delay?: number;
  invertValues?: boolean;
}
