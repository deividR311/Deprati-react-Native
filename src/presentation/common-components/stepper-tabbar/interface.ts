import {
  BottomTabBarProps,
  BottomTabNavigationOptions
} from '@react-navigation/bottom-tabs';
import { ParamListBase, TabNavigationState } from '@react-navigation/native';

export interface PropsButtonTab {
  arrowIconDirection: 'left' | 'right';
  options?: BottomTabNavigationOptions;
  isFocused?: boolean;
  onPress?: () => void;
  onLongPress?: () => void;
}

export interface StepperTabBarNavigationProps extends BottomTabBarProps {
  title?: string;
  subtitle?: string;
  showMenu?: () => void;
  onBackFromTabbar?(state: TabNavigationState<ParamListBase>): void;
  onNextFromTabbar?(state: TabNavigationState<ParamListBase>): void;
  showButtomsWhenZero?: boolean;
}
