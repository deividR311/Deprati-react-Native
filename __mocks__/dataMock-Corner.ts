export const DataMock_CornerContent = {
  defaultPage: true,
  name: 'Dior ',
  template: 'CategoryPageTemplate',
  typeCode: 'ContentPage',
  uid: 'dior',
  timestamp: 1679514724774,
  components: {
    DepratiEmarsysDiorCarouselComponent1: {
      uid: 'DepratiEmarsysDiorCarouselComponent1',
      typeCode: 'DepratiEmarsysProductCarouselComponent',
      modifiedTime: '2022-09-15T15:00:55.221-05:00',
      name: 'DepratiEmarsysDiorCarouselComponent1',
      container: 'false',
      limit: '10',
      logic: 'CATEGORY',
      title: 'Novedades',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    DepratiEmarsysDiorCarouselComponent2: {
      uid: 'DepratiEmarsysDiorCarouselComponent2',
      typeCode: 'DepratiEmarsysProductCarouselComponent',
      modifiedTime: '2022-09-15T15:00:54.903-05:00',
      name: 'DepratiEmarsysDiorCarouselComponent2',
      container: 'false',
      limit: '10',
      logic: 'POPULAR',
      title: 'Lo m√°s vendidos',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    DiorSimpleBannerComponent3: {
      uid: 'DiorSimpleBannerComponent3',
      typeCode: 'SimpleBannerComponent',
      modifiedTime: '2022-09-15T15:00:54.876-05:00',
      name: 'SimpleBannerComponent',
      container: 'false',
      external: 'false',
      media: {
        code: '/images/banners/H_DIOR.jpg',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h41/hd8/8975308816414/H_DIOR.jpg',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h41/hd8/8975308816414/H_DIOR.jpg'
      },
      urlLink: 'https://www.deprati.com.ec/fragance',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    DiorSimpleBannerComponent4: {
      uid: 'DiorSimpleBannerComponent4',
      typeCode: 'SimpleBannerComponent',
      modifiedTime: '2022-09-15T15:00:55.198-05:00',
      name: 'SimpleBannerComponent',
      container: 'false',
      external: 'false',
      media: {
        code: '/images/banners/I_DIOR.jpg',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h98/hd8/8975308849182/I_DIOR.jpg',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h98/hd8/8975308849182/I_DIOR.jpg'
      },
      urlLink: 'https://www.deprati.com.ec/skincare',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    DiorSimpleBannerComponent5: {
      uid: 'DiorSimpleBannerComponent5',
      typeCode: 'SimpleBannerComponent',
      modifiedTime: '2022-09-15T15:00:54.848-05:00',
      name: 'SimpleBannerComponent',
      container: 'false',
      external: 'false',
      media: {
        code: '/images/banners/J_DIOR.jpg',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/ha9/hdb/8975308914718/J_DIOR.jpg',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/ha9/hdb/8975308914718/J_DIOR.jpg'
      },
      urlLink: 'https://www.deprati.com.ec/dior_expertise',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    DiorSimpleBannerComponent1: {
      uid: 'DiorSimpleBannerComponent1',
      typeCode: 'SimpleBannerComponent',
      modifiedTime: '2022-08-12T11:50:29.485-05:00',
      name: 'SimpleBannerComponent',
      container: 'false',
      external: 'false',
      media: {
        code: '/images/banners/B_DIOR.png',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h1c/h11/8977632657438',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h1c/h11/8977632657438'
      },
      urlLink: 'https://www.deprati.com.ec/dior',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    DiorSimpleBannerComponent2: {
      uid: 'DiorSimpleBannerComponent2',
      typeCode: 'SimpleBannerComponent',
      modifiedTime: '2022-09-15T09:41:46.513-05:00',
      name: 'SimpleBannerComponent',
      container: 'false',
      external: 'false',
      media: {
        code: '/images/banners/C_DIOR.jpg',
        mime: 'image/jpeg',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h86/h11/8977632690206',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h86/h11/8977632690206'
      },
      urlLink: '/c/04100750101?q=%3Arelevance%3AFeature-Fragancias%3AJadore',
      childrenComponentsList: null,
      childrenComponents: {}
    },
    DiorCategoryFeatureCarousel1: {
      uid: 'DiorCategoryFeatureCarousel1',
      typeCode: 'DepratiCategoryFeatureCarouselComponent',
      modifiedTime: '2022-09-15T14:44:07.197-05:00',
      name: 'DiorCategoryFeatureCarousel1',
      container: 'false',
      categoryList:
        'DiorCarouselFeatureComponent_1 DiorCarouselFeatureComponent_2 DiorCarouselFeatureComponent_3 DiorCarouselFeatureComponent_4',
      childrenComponentsList: [
        'DiorCarouselFeatureComponent_1',
        'DiorCarouselFeatureComponent_2',
        'DiorCarouselFeatureComponent_3',
        'DiorCarouselFeatureComponent_4'
      ],
      childrenComponents: {
        DiorCarouselFeatureComponent_1: {
          uid: 'DiorCarouselFeatureComponent_1',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2022-09-15T14:59:19+0000',
          name: 'CategoryFeatureComponent',
          container: 'false',
          media: {
            code: '/images/banners/D1_DIOR.jpg',
            mime: 'image/jpeg',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h40/h25/8981861892126',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/h40/h25/8981861892126'
          },
          category: '04100750101',
          title: 'FRAGANCIAS FEMENINAS',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        DiorCarouselFeatureComponent_2: {
          uid: 'DiorCarouselFeatureComponent_2',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2022-09-15T14:59:19+0000',
          name: 'CategoryFeatureComponent',
          container: 'false',
          media: {
            code: '/images/banners/D2_DIOR.png',
            mime: 'image/jpeg',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/he6/h27/8981861924894',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/he6/h27/8981861924894'
          },
          category: '04100750102',
          title: 'FRAGANCIAS MASCULINAS',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        DiorCarouselFeatureComponent_3: {
          uid: 'DiorCarouselFeatureComponent_3',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2022-09-15T14:59:19+0000',
          name: 'CategoryFeatureComponent',
          container: 'false',
          media: {
            code: '/images/banners/D3_DIOR.png',
            mime: 'image/jpeg',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h3e/h28/8981861957662',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/h3e/h28/8981861957662'
          },
          category: '04100750104',
          title: 'CUIDADO FACIAL',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        DiorCarouselFeatureComponent_4: {
          uid: 'DiorCarouselFeatureComponent_4',
          typeCode: 'CategoryFeatureComponent',
          modifiedTime: '2022-09-15T14:59:19+0000',
          name: 'CategoryFeatureComponent',
          container: 'false',
          media: {
            code: '/images/banners/D4_DIOR.png',
            mime: 'image/jpeg',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/ha8/h28/8981861990430',
            downloadUrl:
              'https://imagestg.deprati.com.ec/sys-master/images/ha8/h28/8981861990430'
          },
          category: '04100750103',
          title: 'MAQUILLAJE',
          childrenComponentsList: null,
          childrenComponents: {}
        }
      }
    },
    DiorCategoryDoubleBanner1: {
      uid: 'DiorCategoryDoubleBanner1',
      typeCode: 'DepratiDoubleResponsiveBannerComponent',
      modifiedTime: '2022-09-15T14:44:06.62-05:00',
      name: 'DiorCategoryDoubleBanner1',
      firstBannersList: 'DiorBannerContainer1 DiorBannerContainer2',
      container: 'false',
      childrenComponentsList: ['DiorBannerContainer1', 'DiorBannerContainer2'],
      childrenComponents: {
        DiorBannerContainer1: {
          uid: 'DiorBannerContainer1',
          typeCode: 'SimpleResponsiveBannerComponent',
          modifiedTime: '2022-09-15T14:59:19+0000',
          name: 'SimpleResponsiveBannerComponent',
          container: 'false',
          media: {
            imageType: 'GALLERY',
            format: 'mobile',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/h95/h17/8977632821278',
            width: '480'
          },
          urlLink:
            '/c/04100750101?q=%3Arelevance%3AFeature-Fragancias%3AJadore',
          childrenComponentsList: null,
          childrenComponents: {}
        },
        DiorBannerContainer2: {
          uid: 'DiorBannerContainer2',
          typeCode: 'SimpleResponsiveBannerComponent',
          modifiedTime: '2022-09-15T14:59:19+0000',
          name: 'SimpleResponsiveBannerComponent',
          container: 'false',
          media: {
            imageType: 'GALLERY',
            format: 'mobile',
            url: 'https://imagestg.deprati.com.ec/sys-master/images/hec/h17/8977632854046',
            width: '480'
          },
          urlLink:
            '/c/04100750101?q=%3Arelevance%3AFeature-Fragancias%3AJadore',
          childrenComponentsList: null,
          childrenComponents: {}
        }
      }
    },
    SiteLogoComponent: {
      uid: 'SiteLogoComponent',
      typeCode: 'SimpleBannerComponent',
      modifiedTime: '2022-05-20T11:47:01.945-05:00',
      name: 'Site Logo Component',
      container: 'false',
      external: 'false',
      media: {
        code: '/images/logo_deprati_responsive.svg',
        mime: 'image/svg+xml',
        altText: 'Deprati Storefront',
        url: 'https://imagestg.deprati.com.ec/sys-master/images/h9f/hfd/8969934569502',
        downloadUrl:
          'https://imagestg.deprati.com.ec/sys-master/images/h9f/hfd/8969934569502'
      },
      urlLink: '/',
      childrenComponentsList: null,
      childrenComponents: {}
    }
  },
  slots: {
    Section2: [
      'DepratiEmarsysDiorCarouselComponent1',
      'DepratiEmarsysDiorCarouselComponent2',
      'DiorSimpleBannerComponent3',
      'DiorSimpleBannerComponent4',
      'DiorSimpleBannerComponent5'
    ],
    Section1: [
      'DiorSimpleBannerComponent1',
      'DiorSimpleBannerComponent2',
      'DiorCategoryFeatureCarousel1',
      'DiorCategoryDoubleBanner1'
    ],
    SiteLogo: ['SiteLogoComponent'],
    SearchBox: [],
    NavigationBar: [],
    TopHeaderSlot: []
  }
};
export const DataMock_CornerParams = { labelOrIdCorner: '/dior' };

export const DataMock_CornerRopaFrio = {
  uid: 'string;',
  typeCode: 'string;',
  modifiedTime: 'string;',
  name: 'qwerty name',
  container: false,
  childrenComponentsList: [],
  childrenComponents: {},
  title: 'qwerty',
  style: {},
  width: 400
};
