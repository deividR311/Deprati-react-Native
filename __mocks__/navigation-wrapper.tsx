import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
