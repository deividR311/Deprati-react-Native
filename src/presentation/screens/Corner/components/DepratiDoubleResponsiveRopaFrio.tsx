import React, { useEffect, useState } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Divider } from 'react-native-paper';
import {
  COLORS,
  FontStyles,
  MARING_HORIZONTAL
} from '../../../../application/common';
import ExtensionComponent, {
  ExtensionComponentProps
} from '../../../common-components/extension-component';
import layout from '../../../../application/common/layout';
import { useComponentsMutation } from '../../../../infrastucture/apis/contentPage';

interface Props {
  customProps: CustomProp;
}
interface CustomProp extends ExtensionComponentProps {}

const DepratiDoubleResponsiveRopaFrio = (props: Props) => {
  const { customProps } = props;
  const {
    childrenComponentsList,
    childrenComponents = {},
    title,
    style: customStyle,
    width: widthCustom
  } = customProps;
  const widthScren = widthCustom ?? layout.window.width;
  const [banners, setBanners] = useState([]);
  const [getComponent] = useComponentsMutation();

  const getBanners = async () => {
    try {
      const [nameChild] = childrenComponentsList ?? [];
      const firtsBanner = childrenComponents[nameChild];
      if (customProps?.secondBannersList) {
        const response = await getComponent({
          componentIds: [customProps?.secondBannersList]
        });
        if (response?.data?.component?.length > 0) {
          const [secondBanners] = response?.data?.component;
          setBanners([firtsBanner, secondBanners]);
        }
        return;
      }

      setBanners([firtsBanner]);
    } catch (error) {}
  };

  useEffect(() => {
    if (childrenComponentsList?.length > 0) {
      getBanners();
    }
  }, [childrenComponentsList?.length]);

  if (banners?.length === 0) return null;

  return (
    <View style={[styles.container, customStyle?.container]}>
      <Divider
        style={[
          styles.container__divider_top,
          customStyle?.container__divider_top
        ]}
      />
      {title && (
        <View style={[styles.container__title, customStyle?.container__title]}>
          <Text
            style={[
              styles.container__title_text,
              customStyle?.container__title_text
            ]}>
            {title}
          </Text>
        </View>
      )}
      <View style={[styles.content_banners, customStyle?.content_banners]}>
        {banners?.map((banner, index) => {
          if (banner?.uid)
            return (
              <ExtensionComponent
                key={index}
                {...banner}
                style={{
                  container: {
                    ...styles.banner__container,
                    width: widthScren / 2 - 5
                  },
                  image: styles.banner__container_img
                }}
              />
            );
        })}
      </View>

      <Divider
        style={[
          styles.container__divider_bottom,
          customStyle?.container__divider_bottom
        ]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    marginTop: MARING_HORIZONTAL,
    marginLeft: MARING_HORIZONTAL,
    marginBottom: 10
  },
  container__title: {
    backgroundColor: COLORS.WHITE
  },
  container__title_text: {
    ...FontStyles.Body_2,
    color: COLORS.DARK70
  },
  content_banners: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 2
  },
  banner__container: {
    borderRadius: 4,
    overflow: 'hidden',
    width: '50%'
  },
  banner__container_img: {
    width: '100%'
  },
  container__divider_top: {},
  container__divider_bottom: {}
});

export default DepratiDoubleResponsiveRopaFrio;
