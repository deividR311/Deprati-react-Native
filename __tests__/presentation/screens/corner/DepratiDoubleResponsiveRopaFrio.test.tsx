import React from 'react';
import { render } from '@testing-library/react-native';
import { IRender } from '../../../../__mocks__/IRender.interface';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import DepratiDoubleResponsiveRopaFrio from '../../../../src/presentation/screens/Corner/components/DepratiDoubleResponsiveRopaFrio';
import { DataMock_CornerRopaFrio } from '../../../../__mocks__/dataMock-Corner';

let screenTest: IRender;
describe('CMSParagraph', () => {
  it('renders correctly CMSParagraph in Snapshot', () => {
    screenTest = render(
      <NavigationWrapper
        screen={() => (
          <DepratiDoubleResponsiveRopaFrio
            customProps={DataMock_CornerRopaFrio}
          />
        )}
      />
    );
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
