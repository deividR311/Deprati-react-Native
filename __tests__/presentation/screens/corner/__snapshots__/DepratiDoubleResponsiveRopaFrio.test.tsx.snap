// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CMSParagraph renders correctly CMSParagraph in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
